behaviour/graql/language/delete.feature

```diff
@@ -16,12 +33,566 @@
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value     |
+      | P1   | key   | ref:0     |
+      | P2   | key   | ref:1     |
+      | FR   | key   | ref:2     |
+
+    Then graql delete
+      """
+      match
+        $x isa person, has ref 0;
+        $y isa person, has ref 1;
+      delete
+        $x isa person;
+        $y isa person;
+      """
+
+    Then get answers of graql query
+      """
+      match $r isa friendship; get;
+      """
+    Then answer size is: 0
+
+
+  Scenario: delete a role player with too-specific (downcasting) role throws
+    Given graql define
+      """
+      define
+      special-friendship sub friendship,
+        relates special-friend as friend;
+      """
+    Given the integrity is validated
+
+    When graql insert
+      """
+      insert
+      $x isa person, has ref 0;
+      $y isa person, has ref 1;
+      $r (friend: $x, friend: $y) isa friendship, has ref 2;
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value     |
+      | P1   | key   | ref:0     |
+      | P2   | key   | ref:1     |
+      | FR   | key   | ref:2     |
+
+    Then graql delete throws
+      """
+      match
+        $x isa person, has ref 0;
+        $y isa person, has ref 1;
+        $r (friend: $x, friend: $y) isa friendship;
+      delete
+        $r (special-friend: $x);
+      """
+
+
+  Scenario: delete attribute ownership makes attribute invisible to owner
+    When graql insert
+      """
+      insert
+      $x isa person,
+        has name "john",
+        has ref 0;
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value      |
+      | P1   | key   | ref:0      |
+      | JOHN | value | name:john  |
+
+    Then graql delete
+      """
+      match
+        $x isa person, has name $n;
+        $n "john";
+      delete
+        $x has name $n;
+      """
+
+    Then get answers of graql query
+      """
+      match $x isa person; get;
+      """
+    Then uniquely identify answer concepts
+      | x   |
+      | P1  |
 
-  Scenario: deleted role player is not part of relation
+    Then get answers of graql query
+      """
+      match $n isa name; get;
+      """
+    Then uniquely identify answer concepts
+      | n     |
+      | JOHN  |
 
-  Scenario: deleted attribute not visible from any owner of attribute
+    Then get answers of graql query
+      """
+      match $x isa person, has name $n; get;
+      """
+    Then answer size is: 0
 
-  Scenario: deleted attribute owning instance not visible from attributes
 
-  Scenario: deleting attribute ownership retains attribute and owner but no mutual visibility
+  Scenario: using unmatched variable in delete throws even without data
+    Then graql delete throws
+      """
+      match $x isa person; delete $n isa name;
+      """
```

Comment:
In the case that we can delete attribute ownership we should have a scenario (or more) for deleting shared attributes, ensuring that the other owners are unaffected. 
If we are to release then we need to test that this can be achieved presently, presumably by deleting implicit relations. Depends if they are gone by next release. Either way, we could have the tests and `@ignore` them. 
Also we should have a scenario for deleting the last ownership of an attribute - is the attribute itself queryable or not after that?

Reply:
aight